### PR TITLE
Port XPU for test_masked.py

### PR DESCRIPTION
**Pull request by libohao1201**
For https://github.com/pytorch/pytorch/issues/114850, we will port masked-op test cases to Intel GPU. We could enable Intel GPU with the following methods and try the best to keep the original code styles:

- Instantiate device-type tests with XPU enabled via `instantiate_device_type_tests()`

- Add targeted skips with `skipOps` for known XPU failures

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### Summary

Skip test_mask_layout for masked.prod with complex64/complex128 on XPU, where scatter_gather_base_kernel_func is not yet implemented for complex types in the XPU backend. Also adds allow_xpu=True to instantiate_device_type_tests to properly enable XPU test variants. 

### Changes

- Import skipOps and skip from common_device_type and add @skipOps on test_mask_layout to skip masked.prod for complex64/complex128 on XPU.
- Add allow_xpu=True to instantiate_device_type_tests so XPU test variants are properly instantiated and the skips take effect.

### Test Plan
`pytest -v test/test_masked.py  `

### CODE REVIEW

File: test/test_masked.py [hunk 1]
```diff
@@ -298,6 +298,10 @@ def test_reference_masked(self, device, dtype, op):
     @suppress_warnings
     @ops(masked_ops_with_non_strided_support)
     @precisionOverride({torch.bfloat16: 5e-3, torch.float16: 5e-3})
+    @skipOps([
```
Comment (guangyey) [inline test/test_masked.py]: No, we should implement it on XPU, not skip it.

Comment (CuiYifeng) [reply test/test_masked.py]: @guangyey @libohao1201 Please take a look https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/opinfo/definitions/_masked.py#L562-L569. How about updating opinfo before implementation is ready?

Comment (libohao1201) [reply test/test_masked.py]: Changes in c0d1ca22478c6b6007681bc38ac40df47d70066f.

File: torch/testing/_internal/opinfo/definitions/_masked.py [hunk 2]
```diff
@@ -567,6 +567,14 @@ def sample_inputs_masked_normalize(op_info, device, dtype, requires_grad, **kwar
                 device_type="cuda",
                 dtypes=(torch.bool, *integral_types(), *complex_types()),
             ),
+            # FIXME: scatter_gather_base_kernel_func not implemented for ComplexFloat/ComplexDouble on XPU (sparse_coo inputs)
+            DecorateInfo(
+                unittest.skip("Skipped!"),
+                "TestMasked",
+                "test_mask_layout",
+                device_type="xpu",
+                dtypes=(torch.complex64, torch.complex128),
```
Comment (guangyey) [inline torch/testing/_internal/opinfo/definitions/_masked.py:576]:
```suggestion
                dtypes=(*complex_types()),
```

**Review by guangyey — APPROVED** [review]
LGTM, thanks for your update!

**Review by CuiYifeng — APPROVED** [review]
